packages/exhibitor/extra/start_exhibitor.py

```diff
@@ -175,7 +175,24 @@ def invoke_detect_ip():
     print("ERROR: No known exhibitor backend:", exhibitor_backend)
     sys.exit(1)
 
+truststore_path = '/var/lib/dcos/exhibitor-tls-artifacts/truststore.jks'
+clientstore_path = '/var/lib/dcos/exhibitor-tls-artifacts/clientstore.jks'
+serverstore_path = '/var/lib/dcos/exhibitor-tls-artifacts/serverstore.jks'
+
+exhibitor_env = os.environ.copy()
+if os.path.exists(truststore_path) and \
+   os.path.exists(clientstore_path) and \
+   os.path.exists(serverstore_path):
+    exhibitor_env['EXHIBITOR_TLS_TRUSTSTORE_PATH'] = truststore_path
+    exhibitor_env['EXHIBITOR_TLS_TRUSTSTORE_PASSWORD'] = 'changeme'
+    exhibitor_env['EXHIBITOR_TLS_CLIENT_KEYSTORE_PATH'] = clientstore_path
+    exhibitor_env['EXHIBITOR_TLS_CLIENT_KEYSTORE_PASSWORD'] = 'changeme'
+    exhibitor_env['EXHIBITOR_TLS_SERVER_KEYSTORE_PATH'] = serverstore_path
+    exhibitor_env['EXHIBITOR_TLS_SERVER_KEYSTORE_PASSWORD'] = 'changeme'
+    exhibitor_env['EXHIBITOR_TLS_REQUIRE_CLIENT_CERT'] = 'true'
+    exhibitor_env['EXHIBITOR_TLS_VERIFY_PEER_CERT'] = 'true'
+
 # Start exhibitor
 print("Running exhibitor as command:", exhibitor_cmdline)
 sys.stdout.flush()
-os.execv('/opt/mesosphere/bin/java', exhibitor_cmdline)
+os.execve('/opt/mesosphere/bin/java', exhibitor_cmdline, exhibitor_env)
```

Comment:
Interesting. You could also modify the current environment (the environment of the CPython process by mutating `os.environ` directly, and then keep `os.execv()` as-is.

Do you prefer one of the other and if yes, why?

Reply:
I would prefer to pass the new environment variables explicitly to the new process. Since the parent process does not need them why change its environment (even though it will be cleanly replaced by it's child).

Reply:
The parent process' purpose is as a launcher for the child, I think changing the parent's environment in that case makes complete sense. I wouldn't block this PR on that change though.